### ISSUE TITLE
SSH server does not work when running as windows service.

### ISSUE THREAD

**Issue by btrepp** [go-gitea/gitea#680]
- Gitea version (or commit ref): (my own fork, but should happen on all versions)
- Git version: 2.11.0-32
- Operating system: Windows
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] SQLite
  - [x] MSSQL
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description

When running the inbuilt SSH server, ssh fails. My investigation has shown this to be caused by how minwinsvc determines is is running in service mode. What happens is as follows.

The ssh server starts correctly, and attempts to run the gitea.exe executable with the serv argument when a users is correctly authenticated with a SSH key (I have verified it is picking up keys correctly)
The gitea serv command returns 0 with no console output when run this way (via the inbuilt ssh server).

This is an issue with how minwinsvc determines whether to run as a service or console, the init code for minwinsvc calls Os exit with 0 if ``IsAnInteractiveSession()`` returns false. This is causing the shelling out of gitea to calling itself to return instantly.

I have created a ticket in https://github.com/kardianos/minwinsvc/issues/2. 

NOTE: The server works fine in windows when run via a console. As IsAnInteractiveSession will return true.

TLDR;. on windows using in built ssh server and running as a service; gitea.exe thinks it is a service when executing `git-recieve-pack` via ssh, and doesn't write to stdout, thus closing the connection to the client.

...

**Comment by andreynering** [comment]
I think the only way is to hack with minwinsvc's internals to prevent that.

@btrepp Since I don't usually use the SSH server, can you help me test this patch?

https://gist.githubusercontent.com/andreynering/90747602e75a33e3b69df793d4171250/raw/245eb559c1dd3b0cc95659ac633bdb8c14ecaf8e/gites-windows-ssh.patch

**Comment by btrepp** [comment]
@andreynering the patch works well, applied to current master and am running it on a testing machine as a windows service. 

Successfully pushed giteas source into my testing machine, and cloned it out elsewhere :)

**Comment by andreynering** [comment]
@btrepp Awesome! I'll open PR when possible.
